sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1507,7 +1507,8 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         case "TIMESTAMP" =>
           Literal(Timestamp.valueOf(value))
         case "X" =>
-          val padding = if (value.length % 2 == 1) "0" else ""
+          // avoid false positive of IM_BAD_CHECK_FOR_ODD by SpotBugs
```

Comment:
what does it mean?

Reply:
[It](http://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#im-check-for-oddness-that-won-t-work-for-negative-numbers-im-bad-check-for-odd) means `x % 2 == ` does not work correctly if x is negative.

In this case, we know `value.length` is always non-negative . The current SpotBugs cannot detect this pre-condition. I cannot find to disable only this check in `findPuzzles`, too.

Therefore, this is a workaround to avoid the false positive violation.